lib/services/identity.go

```diff
@@ -40,9 +40,12 @@ type UserGetter interface {
 	GetUser(user string, withSecrets bool) (User, error)
 }
 
-// UserService is reponsible for basic user management
+// UsersService is reponsible for basic user management
 type UsersService interface {
 	UserGetter
+	// UpsertUserV2 updates an existing user or creates a new user entry in a backend.
+	// Returns a bool that indicates if user existed or not.
+	UpsertUserV2(ctx context.Context, user User) (bool, error)
 	// UpsertUser updates parameters about user
 	UpsertUser(user User) error
```

Comment:
Clarify the difference between these in a comment.
Should `UpsertUser` be called at all anymore?